### PR TITLE
[ebpf] Add `ebpf.Modifier` for removing helper calls

### PR DESCRIPTION
**Pull request by p-lambert**
### What does this PR do?

Add `ebpf.Modifer` for removing certain helper calls from eBPF bytecode.

### Motivation

Allow eBPF developers to write code referencing multiple eBPF helpers that get "conditionally selected" during runtime.
For example, with this change someone might write code such as the following:

```
if (ring_buffers_supported) {
    bpf_ringbuf_output(...);
} else {
    bpf_perf_event_output(...);
}
```

And then use this `ebpf.Modifer` to strip out the call to `bpf_ringbuf_output` in case the target kernel doesn't support it:

```
if (ring_buffers_supported) {
    // NO-OP
} else {
    bpf_perf_event_output(...);
}
```

### Additional Notes

Please try to favor the use of either CO-RE or runtime compilation to conditionally select eBPF helpers. This should be regarded as a last resort when the aforementioned options don't apply (prebuilt artifacts, for example).

### CODE REVIEW

File: pkg/ebpf/helper_call_patcher.go [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+			for iter.Next() {
+				ins := iter.Ins
+				if !ins.IsBuiltinCall() {
+					continue
+				}
+
+				for _, fn := range h.helpers {
+					if ins.Constant == int64(fn) {
+						*ins = noopIns.WithMetadata(ins.Metadata)
+						continue InstructionLoop
```
Comment (brycekahle) [inline pkg/ebpf/helper_call_patcher.go:61]: `break` (without a label) would be equivalent here, correct?

Comment (p-lambert) [reply pkg/ebpf/helper_call_patcher.go:61]: Good point. I'll update it

**Review by brycekahle — APPROVED** [review]

**Pull request merged**